ncm-dpmlfc/src/main/perl/dpmlfc/RuleBasedEditor.pm

```diff
@@ -0,0 +1,817 @@
+    unless ( $line_fmt ) {
+      $line_fmt = $LINE_FORMAT_DEFAULT;
+    }
+    my $value_opt;
+    if ( $value_fmt ) {
+      ($value_fmt, $value_opt) = split /:/, $value_fmt;
+    }else {
+      $value_fmt = LINE_VALUE_AS_IS;
+    }
+    unless ( defined($value_opt) ) {
+      $value_opt = LINE_VALUE_OPT_NONE;      
+    }
+
+    (my $condition, my $tmp) = split /->/, $rule;
+    if ( $tmp ) {
+      $rule = $tmp;
+    } else {
+      $condition = "";
+    }
+    $self->debug(1,"$function_name: processing rule $rule_id (variable=>>>$keyword<<<, comment_line=",
+                          "$comment_line, condition=>>>$condition<<<, rule=>>>$rule<<<, fmt=$line_fmt)");
+
+    # If the keyword was "negated", remove (comment out) configuration line if present and enabled
+    if ( $comment_line ) {
+      $self->debug(1,"$function_name: keyword '$keyword' negated, removing configuration line");
+      $self->_removeConfigLine($fh,$keyword,$line_fmt);
+      next;
+    }
+
+    # Check if only rules with ALWAYS condition must be applied.
+    # ALWAYS is a special condition that is used to flag the only rules that
+    # must be applied if the option always_rules_only is set. When this option
+    # is not set, this condition has no effect and is just reset to an empty conditions.
+    if ( $parser_options->{always_rules_only} ) {
+      if ( $condition ne $RULE_CONDITION_ALWAYS ) {
+        $self->debug(1,"$function_name: rule ignored ($RULE_CONDITION_ALWAYS condition not set)");
+        next;
+      }
+    }
+    if ( $condition eq $RULE_CONDITION_ALWAYS ) {
+      $condition = '';
+    }
+
+    # Check if rule condition is met if one is defined
+    unless ( $condition eq "" ) {
+      $self->debug(1,"$function_name: checking condition >>>$condition<<<");
+
+      # Condition may be negated if it starts with a !: remove it from the condition value.
+      # If the condition is negated, when the condition is true the rule must not be applied.
+      my $negate = 0;
+      if ( $condition =~ /^!/ ) {
+        $negate = 1;
+        $condition =~ s/^!//;
+      }
+      my ($cond_attribute,$cond_option_set) = split /:/, $condition;
+      unless ( $cond_option_set ) {
+        $cond_option_set = $cond_attribute;
+        $cond_attribute = "";
+      }
+      $self->debug(2,"$function_name: condition option set = '$cond_option_set', ".
+                         "condition attribute = '$cond_attribute', negate=$negate");
+      my $cond_satisfied = 1;       # Assume condition is satisfied
+      if ( $cond_attribute ) {
+        # Due to Perl autovivification, testing directly exists($config_options->{$cond_option_set}->{$cond_attribute}) will spring
+        # $config_options->{$cond_option_set} into existence if it doesn't exist.
+        my $cond_true = exists($config_options->{$cond_option_set}) && 
+                        exists($config_options->{$cond_option_set}->{$cond_attribute});          
+        if ( $negate ) {
```

Comment:
maybe rewrite this to 
```perl
$cond_true = ! $cond_true if $negate
$cond_satisfied = 0 if $cond_true;
```
(altough this changes `$cond_true`, don't know if it is used below)